test/suites/device-snmp/main_test.go

```diff
@@ -0,0 +1,65 @@
+package test
+
+import (
+	"edgex-snap-testing/test/utils"
+	"log"
+	"os"
+	"testing"
+)
+
+const deviceSnmpSnap = "edgex-device-snmp"
+const deviceSnmpService = "edgex-device-snmp.device-snmp"
+
+var platformPorts = []string{
+	"59880", // core-data
+	"59881", // core-metadata
+	"59882", // core-command
+	"8000",  // kong
+	"5432",  // kong-database
+	"8200",  // vault
+	"8500",  // consul
+	"6379",  // redis
+}
+
+func TestMain(m *testing.M) {
+
+	log.Println("[SETUP]")
+
+	// start clean
+	utils.SnapRemove(nil,
+		deviceSnmpSnap,
+		"edgexfoundry",
+	)
+
+	// install the device snap before edgexfoundry
+	// to catch build error sooner and stop
+	if utils.LocalSnap != "" {
+		utils.SnapInstallFromFile(nil, utils.LocalSnap)
+	} else {
+		utils.SnapInstallFromStore(nil, deviceSnmpSnap, utils.ServiceChannel)
+	}
+	utils.SnapInstallFromStore(nil, "edgexfoundry", utils.PlatformChannel)
+
+	// make sure all services online before starting subtests
```

Comment:
Note that `Test*` are tests and `t.Run("*",..` inside those tests are subtests. There could also be sub-subtests i.e. t.Run inside another t.Run.